src/ngx_http_lua_ssl_session_fetchby.c

```diff
@@ -597,8 +598,13 @@ ngx_http_lua_ffi_ssl_set_serialized_session(ngx_http_request_t *r,
         return NGX_ERROR;
     }
 
+    old_session = cctx->session;
     cctx->session = session;
 
+    if (old_session) {
```

Comment:
We should test a pointer as a pointer instead of as a boolean. The recommended way is to write `old_session != NULL` here. I fixed it in the merged version.